rllib/core/examples/simple_ppo_rl_module.py

```diff
@@ -0,0 +1,321 @@
+            }
+        )
+
+    @override(RLModule)
+    def _forward_inference(self, batch: NestedDict) -> Mapping[str, Any]:
+        if self.encoder:
+            encoded_state = self.encoder(batch["obs"])
+            pi_out = self.pi(encoded_state)
+        else:
+            pi_out = self.pi(batch["obs"])
+
+        if self._is_discrete:
+            action = torch.argmax(pi_out, dim=-1)
+        else:
+            action, _ = pi_out.chunk(2, dim=-1)
+
+        action_dist = TorchDeterministic(action)
+        return {"action_dist": action_dist}
+
+    @override(RLModule)
+    def input_specs_exploration(self):
+        return ModelSpec(
+            {
+                "obs": self.encoder.input_specs()
+                if self.encoder
+                else self.pi.input_specs(),
+            }
+        )
+
+    @override(RLModule)
+    def output_specs_exploration(self) -> ModelSpec:
+        return ModelSpec(
+            {
+                "action_dist": TorchCategorical
+                if self._is_discrete
+                else TorchDiagGaussian,
+            }
+        )
+
+    @override(RLModule)
+    def _forward_exploration(self, batch: NestedDict) -> Mapping[str, Any]:
+        if self.encoder:
+            encoded_state = self.encoder(batch["obs"])
+            pi_out = self.pi(encoded_state)
+        else:
+            pi_out = self.pi(batch["obs"])
+
+        if self._is_discrete:
+            action_dist = TorchCategorical(pi_out)
+        else:
+            mu, scale = pi_out.chunk(2, dim=-1)
+            action_dist = TorchDiagGaussian(mu, scale.exp())
+
+        return {"action_dist": action_dist}
+
+    @override(RLModule)
+    def input_specs_train(self) -> ModelSpec:
+        if self._is_discrete:
+            action_dim = 1
+        else:
+            action_dim = self.config.action_space.shape[0]
+
+        return ModelSpec(
+            {
+                "obs": self.encoder.input_specs()
+                if self.encoder
+                else self.pi.input_specs(),
+                "action": TorchTensorSpec("b, da", da=action_dim),
+            }
+        )
+
+    @override(RLModule)
+    def output_specs_train(self) -> ModelSpec:
+        return ModelSpec(
+            {
+                "action_dist": TorchCategorical
+                if self._is_discrete
+                else TorchDiagGaussian,
+                "logp": TorchTensorSpec("b", dtype=torch.float32),
+                "entropy": TorchTensorSpec("b", dtype=torch.float32),
+                "vf": TorchTensorSpec("b", dtype=torch.float32),
+            }
+        )
+
+    @override(RLModule)
+    def _forward_train(self, batch: NestedDict) -> Mapping[str, Any]:
+        if self.encoder:
+            encoded_state = self.encoder(batch["obs"])
+            pi_out = self.pi(encoded_state)
+            vf = self.vf(encoded_state)
+        else:
+            pi_out = self.pi(batch["obs"])
+            vf = self.vf(batch["obs"])
```

Comment:
nit, simpler if we write:
```
encoded_state = batch["obs"]
if self.encoder:
    encoded_state = self.encoder(encoded_state)
pi_out = self.pi(encoded_state)
vf = self.vf(encoded_state)
```
btw, all these ``"obs"`` should probably be ``SampleBatch.OBS``?

Reply:
fixed the simpler implementation. regarding the SampleBatch.OBS thingy, actually no, not here. This is written without any assumption on existence of a `SampleBatch` class, so it should ideally not import it.